cluster_experiments/perturbator.py

```diff
@@ -90,6 +91,206 @@ def perturbate(
         return df
 
 
+class StochasticPerturbator(Perturbator):
+    """The StochasticPerturbator class implements a perturbator that adds a stochastic effect
+    to the target column of the treated instances. The stochastic effect is sampled from a
+    normal distribution with mean average_effect and variance scale. If scale is not
+    provided, the variance is abs(average_effect).
+
+    Arguments:
+        average_effect (Optional[float], optional): the average effect of the treatment. Defaults to None.
+        target_col (str, optional): name of the target_col to use as the outcome. Defaults to "target".
+        treatment_col (str, optional): the name of the column that contains the treatment. Defaults to "treatment".
+        treatment (str, optional): name of the treatment to use as the treated group. Defaults to "B".
+        scale (Optional[float], optional): the scale of the effect distribution. Defaults to None.
+    """
+
+    def __init__(
```

Comment:
I think you need to add a from_config method and pass the scale in there, if not it won't be sent when building from config

Reply:
nice catch! it's fixed now